File: backend/app/generate_entity_hash.py
import hashlib
import random
import string

def generate_hash(name: str) -> str:
    initials = ''.join([word[0].upper() for word in name.split()])
    random_part = ''.join(random.choices(string.ascii_letters + string.digits, k=4))

    name_hash = hashlib.sha256(name.encode()).hexdigest()[:8]
    entity_hash = f"{initials}{random_part}{name_hash}"
    return entity_hash

if __name__ == "__main__":
    name = "Jason Mraz Kaleeshi"
    entity_id = generate_hash(name)
    print(f"Entity Hash for '{name}': {entity_id}")

File: frontend/src/utils/generateEntityHash.ts
import crypto from "crypto";

/**
 * Generate a unique hash for an entity based on its name.
 * Mimics the Python version: "Initials + Random Part + Hash"
 */
export const generateEntityHash = (name: string): string => {
  if (!name) return "";

  // Extract initials from name
  const initials = name
    .split(" ")
    .map((word) => word[0].toUpperCase())
    .join("");

  // Generate a random 4-character alphanumeric string
  const randomPart = Math.random().toString(36).substring(2, 6).toUpperCase();

  // Generate an 8-character hash from the name
  const hash = crypto.createHash("sha256").update(name).digest("hex").substring(0, 8);

  // Combine parts to create unique identifier
  return `${initials}${randomPart}${hash}`;
};
